### PR TITLE
docs: [IAC-2094]: IaCM Security documentation

### PR DESCRIPTION
**Pull request by richardblack-Harness**
## Description

* Add new IaCM Security doc with interactive diagram
* [JIRA](https://harness.atlassian.net/browse/IAC-2094)

## Screenshots

https://github.com/harness/developer-hub/assets/167115273/d263986a-9bec-4a25-89ba-606b6eb9fe66

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/infra-as-code-management/iacm_security.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+	1. **Workspace and Configuration Setup:**
+    - Harness IaCM retrieves the workspace configuration and associated files, including dependent IaC modules specified in your settings.
+	2.	**Variable and Secret Integration:**
+	- Variables and secret values defined in your workspace are collected and resolved, adhering to Harness Platform security protocols.
+	3.	**Execution Preparation:**
+	- All relevant configuration files and dependent modules are cloned into the pipeline environment. Secrets are integrated to facilitate interaction with your IaC-managed resources.
+	- IaC applications are pulled to your environment just in time for execution, ensuring that the most up-to-date configurations are applied.
+	4.	**Data Management:**
+	- A read-only copy of essential files such as the plan and state files, is uploaded to Harness Cloud to enable IaCM functionalities.
```
Comment (douglas-j-bothwell) [inline docs/infra-as-code-management/iacm_security.md:41]:
```suggestion
	1. **Workspace and Configuration Setup:**
            - Harness IaCM retrieves the workspace configuration and associated files, including dependent IaC modules specified in your settings.
	2.  **Variable and Secret Integration:**
	     - Variables and secret values defined in your workspace are collected and resolved, adhering to Harness Platform security protocols.
	3. **Execution Preparation:**
	    - All relevant configuration files and dependent modules are cloned into the pipeline environment. Secrets are integrated to facilitate interaction with your IaC-managed resources.
	    - IaC applications are pulled to your environment just in time for execution, ensuring that the most up-to-date configurations are applied.
	4. **Data Management:**
	     - A read-only copy of essential files such as the plan and state files, is uploaded to Harness Cloud to enable IaCM functionalities.
```
I'm seeing indentation issues in the published output.

File: docs/infra-as-code-management/iacm_security.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  2. A copy of the plan is passed to provide cost estimation data.
+  3. A copy of the plan is passed to enforce implicit policies set on the Plan File entity.
+  4. IaCM stores a copy of the plan to provide pipeline & historical information on what has changed.
+  5. Apply/Destroy command honours the defined backed & operates with that alone.
+  6. A copy of the state is passed to enforce implicit policies set on the State File entity.
+  7. IaCM stores a copy of the state file to ensure historic state tracking, resource views etc in UI.
+  </TabItem>
+</Tabs>
+
+### Common Security Concerns
```
Comment (douglas-j-bothwell) [inline docs/infra-as-code-management/iacm_security.md:111]: I'm not sure why this content is at the bottom, under a level-3 heading. Consider incorporating this with the introduction at the top.

**Review by douglas-j-bothwell — DISMISSED** [review]
A few sugs, otherwise LGTM

File: docs/infra-as-code-management/iacm_security.md (hunk 2 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/iacm_security.md]: Done

**Review by RickCraig — APPROVED** [review]

File: docs/infra-as-code-management/iacm_security.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+During the planning and execution phases, Harness ensures compliance by checking changes against organizational policies and detecting any tampering with state files before execution. These measures maintain a secure and compliant environment for managing infrastructure changes.
+
+--- 
+The operational model flow is comprised of three components:
+<Tabs>
+<TabItem value="Defined backend">
+All executed commands honor your defined backend, which determines where your infrastructure state is stored and how operations such as `apply` or `destroy` are executed. 
+
+:::note default backend
+If no plan file is specified, the IaCM backend operates as if Terraform is running in `local` mode. For more details, go to [Terraform local backend](https://developer.hashicorp.com/terraform/language/settings/backends/local) or learn how to [initialize your remote backend](https://developer.harness.io/docs/infra-as-code-management/remote-backends/init-configuration).
```
Comment (allofthesepeople) [inline docs/infra-as-code-management/iacm_security.md:32]: We've a ticket to change this behaviour so might leave the detail off "local mode" out.

Maybe something like
```suggestion
If no plan file is specified, IaCM will apply its own backend implicitly.
```
???

**Review by allofthesepeople — COMMENTED** [review]
just 2 nits

File: docs/infra-as-code-management/iacm_security.md (hunk 3 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/iacm_security.md:32]: Yea I was trying to confirm what applied implicitly meant in this context, but that's grand.

**Review by allofthesepeople — APPROVED** [review]

**Pull request merged**